webboot/webboot.go

```diff
@@ -79,18 +79,13 @@ func linkOpen(URL string) (io.ReadCloser, error) {
 	return resp.Body, nil
 }
 
-// write copies the content from an io.readCloser to a named file with filename.
-func write(read io.ReadCloser, filename string) error {
-	w, err := os.Create(filename)
-	if err != nil {
-		return err
-	}
-
-	defer w.Close()
-	defer read.Close()
+// TODO: Find a persistent memory block large enough to store the content. If no blocks are available, error the user.
+// write copies the content from an io.readCloser to a persistent memory block.
+func write(body []byte, pmem *os.File) error {
+	defer pmem.Close()
```

Comment:
put this defer at line 155, not here. The function should not close a file passed as a parameter.

Reply:
That is a good point. I will let main take care of it all since it is opening the file then it should be responsible for closing the file as well, so the logic takes consistent.